mage/query-modules/python/gnn-node-classification.md

```diff
@@ -0,0 +1,343 @@
+
+### About the query module
+
+This query module contains all necessary functions you need to train GNN model on Memgraph. 
+
+The `node_classification` module supports as follows:
+- homogeneous and heterogeneous graphs
+- multiple-label and multi-edge-type graphs
+- any-size datasets
+- the following model architectures:
+    - Graph Attention with Jumping Knowledge
+    - multiple versions of Graph attention networks (GAT)
+    - GraphSAGE
+- early stopping
+- calculation of various metrics
+- predictions for specified nodes
+- model saving and loading
+- recommendation system use cases
+
+
+The easiest way to test **node_classification** is by downloading [Memgraph Platform](https://memgraph.com/download)
+and using some of the preloaded datasets in **Memgraph Lab**. If you want to explore our implementation, jump to **[github/memgraph/mage](https://github.com/memgraph/mage)** and find
+`python/node_classification.py`. Feel free to give us a :star: if you like the code. 
+
+
+Feel free to open a **[GitHub issue](https://github.com/memgraph/mage/issues)**
+or start a discussion on **[Discord](https://discord.gg/memgraph)** if you want
+to speed up development.
+
+## Usage
+Load dataset in Memgraph, call `set_model_parameters`, and start training your model. When training is done, query module will save models. 
+Afterwards, you can test modules on other data (which model has not already seen for example) and inspect the results!
+The module reports the [mean average precision](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html)
+for every batch `training` or `evaluation` epoch.
+
+
+To **summarize** basic node classification workflow is as follows:
+
+- load data to Memgraph
+- set parameters by calling `set_model_parameters()` function. Be sure that **node_features** property on nodes are in place.
+- call `train()` function
+- inspect training results (optional) by calling `get_training_data()` function
+- optionally use `save_model()` and `load_model()` if you want
+- predict node class by calling `predict()` procedure
+
+
+:::info
+
+This **MAGE** module is still in its early stage. We intend to use it only for
+**exploring or learning** about node classification . If you wish us to make it production-ready, make sure
+to either open a **[GitHub issue](https://github.com/memgraph/mage/issues)** or
+drop us a comment on **[Discord](https://discord.gg/memgraph)**. Also, consider
+throwing us a :star: so we can continue to do even better work.
+
+:::
+
+## Procedures
+
+### `set_model_parameters(params)`
+
+This function initializes all global variables. _You_ can change those via **params** dictionary. It checks if variables in **params** are defined appropriately. If so, map of default global parameters is overriden with user defined dictionary params.
+After that it executes previously defined functions declare_globals and
+declare_model_and_data and sets each global variable to some value.
```

Comment:
```suggestion
The function initializes all global variables. _You_ can change global variables via **params** dictionary. It checks if variables in **params** are defined appropriately. If so, map of default global parameters is overriden with user defined dictionary params.
After that it executes previously defined functions declare_globals and
declare_model_and_data and sets each global variable to some value.
```

Reply:
What is IT in that third sentence? Dictionary? 
If so = should this be -> if they are?

And later: after that IT executes - what is IT?